apps/demo/src/cms.tsx

```diff
@@ -10,22 +11,28 @@ const featureSchema = v.object({
   description: v.string(),
 })
 
+const featureResolver = createJsonResolver(
+  contentPaths.feature,
+  featureSchema,
+  {
+    maxBatchSize: 10,
+  },
+)
+
 const textSchema = v.string()
 
+const textResolver = createJsonResolver(contentPaths.text, textSchema, {
+  maxBatchSize: 10,
+})
+
 const contracts = {
-  text: {
-    resolver: createJsonResolver(contentPaths.text, textSchema, {
-      maxBatchSize: 10,
-    }),
-    schema: textSchema,
-  },
-  feature: {
-    resolver: createJsonResolver(contentPaths.feature, featureSchema, {
-      maxBatchSize: 10,
-    }),
-    schema: featureSchema,
-  },
-}
+  text: createContract(textSchema, textResolver),
+  feature: createContract(featureSchema, featureResolver, {
+    name: {
+      rich: true,
+    },
+  }),
+} as unknown as Contracts
```

Comment:
@electric-gizzard the `createContract` helper returns a strongly typed `Contract`, which is great and helpful for ensuring the keys on `meta` are confined to only those defined within the schema. However, that seems to be incompatible with `Contracts` unfortunately. I get [this error](https://ibc-ps.com/).

The definition of `Contracts` right now accepts a base schema, which seems to be the ultimate cause of this issue. The casting here solves the problem, obviously, but it'd be nice to avoid that.

Thoughts?

Reply:
@electric-gizzard I don't know. Open to ideas there... probably just this?

```
{
  rich: true
}
```

instead of

```
{
  [property]: {
    rich: true
  }
}
```

Nested objects could get tricky.

Reply:
@electric-gizzard I'm also open to not structuring the meta schema like this and go with the decorator approach if that's easier.

Reply:
@electric-gizzard do you think the level of effort of the decorator approach is less?

Reply:
We might be able to leverage Valibot's `metadata` action: https://valibot.dev/api/metadata/. Though, that'll couple us more tightly to Valibot. That's probably okay for now.

Reply:
<img width="962" alt="image" src="https://github.com/user-attachments/assets/49300b76-51a8-477f-90b7-af529ab31a80" />

Reply:
Looks like we can change the error mode in the `toJsonSchema` [config](https://github.com/fabian-hiller/valibot/blob/3532a5e71d0a9622ee3bdcf879e7ea77101fbd8d/packages/to-json-schema/src/utils/handleError.ts#L14).